### PR TITLE
WT-4993 Migrate wiredtiger-configure-combinations

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

**Comment by quchenhao** [comment]
evergreen retry

File: configure_combinations [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+sh build_posix/reconf
```
Comment (lukech) [inline configure_combinations]: This line assumes the script will be called under top-level directory of wiredtiger tree. I'd prefer making it portable by querying out and cd-ing to the top-level directory.

File: configure_combinations [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+flags="CFLAGS=\"-Werror -Wall -Wextra -Waddress -Waggregate-return -Wbad-function-cast -Wcast-align -Wdeclaration-after-statement -Wformat-security -Wformat-nonliteral -Wformat=2 -Wmissing-declarations -Wmissing-field-initializers -Wmissing-prototypes -Wnested-externs -Wno-unused-parameter -Wpointer-arith -Wredundant-decls -Wshadow -Wundef -Wunused -Wwrite-strings -O -fno-strict-aliasing -Wuninitialized\"
+CC=clang CFLAGS=\"-Wall -Werror -Qunused-arguments -Wno-self-assign -Wno-parentheses-equality -Wno-array-bounds\""
+
+options="--enable-diagnostic
+--disable-shared
+--disable-static
+--enable-java --enable-python
+--enable-snappy --enable-zlib --enable-lz4
+--with-builtins=lz4,snappy,zlib
+--enable-python
```
Comment (lukech) [inline configure_combinations]: `--enable-python` itself may not deserve a separate test. Suggest to drop this line.

File: configure_combinations [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+insdir=`pwd`/installed
+export PKG_CONFIG_PATH=$insdir/lib/pkgconfig
+IFS="$cr_IFS"
+for flag in $flags ; do
+        for option in $options ; do
+               cd "$curdir"
+               IFS="$saved_IFS"
+               if ! BuildTest "$flag" "$option" "$@"; then
+                       ecode=1
+                       cp test-suite.log test-suite.log.bad
```
Comment (lukech) [inline configure_combinations]: Did not see where `test-suite.log` was written.

File: configure_combinations (hunk 1 above)
Comment (quchenhao) [reply configure_combinations]: done

File: configure_combinations (hunk 2 above)
Comment (quchenhao) [reply configure_combinations]: removed

File: configure_combinations [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        eval ../configure $extra_config "$1" "$2" \
+                 --prefix="$insdir" || return 1
+        eval make "$3" || return 1
+        make -C examples/c check VERBOSE=1 > /dev/null || return 1
+        case "$2" in
+		# Execute python test
+		*with-timestamp-size=0*)
+                    doinstall=false
+                    env LD_LIBRARY_PATH=.libs python ../test/suite/run.py -v 4 "$3"
+                    ;;
```
[the remark this answers is not in the document]
Comment (quchenhao) [reply configure_combinations]: removed

File: configure_combinations (hunk 3 above)
Comment (quchenhao) [reply configure_combinations]: removed

**Pull request merged**